pg_shard--1.0--1.1.sql

```diff
@@ -12,7 +12,16 @@ AS $sync_table_metadata_to_citus$
 		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
 		dummy_shard_length CONSTANT bigint := 0;
 	BEGIN
-		-- copy shard placement metadata
+		-- grab lock for upsert
+		LOCK TABLE pg_dist_shard_placement IN EXCLUSIVE MODE;
+
+		-- perform update of shard health
```

Comment:
As far as I understand, we aim to 
* Firstly update existing shard placements in pg_dist_shard_placement
* Then, add the shard_placements that does not exist in pg_dist_shard_placement,  pg_dist_shard and  pg_dist_partition.

So, updating the comments that explains this could be better, what do you think?
1) -- perform update of shard health => -- perform update of shard health which are already synced
2) -- copy new shard placement metadata => -- copy shard placement metadata which does not exist in pg_dist_shard_placement
3) -- copy new shard metadata  =>  copy shard metadata which does not exist in pg_dist_shard
4) -- copy new partition metadata, which also converts the partition column to ...   => copy partition metadata which does not exist in pg_dist_partition ..